### PR TITLE
Remove unused methods of OC_Response

### PR DESCRIPTION
**Pull request by MorrisJobke**
Ref #8375 and #7827

### CODE REVIEW

File: apps/files_sharing/public.php [hunk 1]
```diff
@@ -28,7 +28,14 @@
 $route = isset($_GET['download']) ? 'files_sharing.sharecontroller.downloadShare' : 'files_sharing.sharecontroller.showShare';
 
 if($token !== '') {
-	OC_Response::redirect($urlGenerator->linkToRoute($route, array('token' => $token)));
+	$protocol = \OC::$server->getRequest()->getHttpProtocol();
+	if ($protocol == 'HTTP/1.1') {
```
Comment (ChristophWurst) [inline apps/files_sharing/public.php]: `===`?

**Review by ChristophWurst — CHANGES_REQUESTED** [review]
Looks good otherwise

File: apps/files_sharing/public.php (hunk 1 above)
Comment (MorrisJobke) [reply apps/files_sharing/public.php]: Just a copy paste and should be replaced soon anyways.

**Review by ChristophWurst — APPROVED** [review]

File: apps/files_sharing/public.php [hunk 2]
```diff
@@ -28,7 +28,14 @@
...
+		$status = '307 Temporary Redirect';
+	} else {
+		$status = '304 Found';
```
Comment (skjnldsv) [inline apps/files_sharing/public.php]: 304 Not Modified is the proper code?

Comment (skjnldsv) [reply apps/files_sharing/public.php]: Or 302 Found? :)

**Comment by rullzer** [comment]
HTTP 1.0 support needs to die anyway. 1.1 was release in 1997

**Review by rullzer — APPROVED** [review]

**Comment by skjnldsv** [comment]
Cant we return HTTP 418 instead? :D

**Pull request merged**